### ISSUE TITLE
Collection populate with references

### ISSUE THREAD

**Issue by Wolvereness** [mikro-orm/mikro-orm#1158]
**Is your feature request related to a problem? Please describe.**
When working with a `ManyToMany` relationship, I only need the primary-key of the corresponding entity. All methods of populating the `Collection` yields the corresponding entities fully populated.

**Describe the solution you'd like**
A method that only accesses the relationship table to make uninitialized references in the `Collection`.

**Describe alternatives you've considered**
Fully populating the entities works, but it's much more data than the problem requires.

**Additional context**
Joining the relationship-table can produce a lot of data in quadratic big-O, and double-select increases the request latency.

**Comment by B4nan** [comment]
It's worth noting that there is actually only one query that loads both the pivot table and joins the target table (also in the select in strategy). There will always be entities - references are still entities. The only things that we can skip by doing this is the join and hydration of target entities. Of course for large collections this can be worth it.

I will also add that in v4 it is possible to add to an uninitialized collection.

Here is a proposed API for this:

```ts
await book.tags.init({ reference: true });
await em.populate(book, ['tags:ref']);
await em.find(Book, { ... }, ['tags:ref']);
// might use something else than `LoadStrategy`, not sure now
await em.find(Book, { ... }, { populate: { tags: LoadStrategy.REFERENCE } });
```

**Comment by olivierlacan** [comment]
@B4nan This is exactly what we were looking for. We'd love to be able to just init the join/pivot table primary key(s).

If you need any help bringing this feature up or testing it, lmk. 😃

**Comment by B4nan** [comment]
I guess the biggest issue right now might be the proposed API, as populate hints are now (v5) string arrays only and they are strictly typed, so we would need to adjust that part to support things like `['tags:ref']` (already tried some time ago, it seemed harder than anticipated) or come up with a different way to pass this information (which would itself be probably a lot of work to support). The implementation itself might be rather easy.

**Comment by B4nan** [comment]
Implemented in v6 via #4776

**Comment by Wolvereness** [comment]
Awesome! Looking forward to our v6 upgrade.
